### PR TITLE
Implemented week recurrence and duplicate value handleing for expense…

### PR DESCRIPTION
**Pull request by kandoeniko**
…s and incomes

### CODE REVIEW

File: UI/RecurringExpenseInput.cs [hunk 1]
```diff
@@ -257,6 +262,14 @@ private int CalcMonthsInRange()
         }
 
         /// <summary>
+        /// Calculates the months in the range from the start date to the end date
```
Comment (M-Zuber) [inline UI/RecurringExpenseInput.cs]: The comments here need to be updated.

File: UI/RecurringExpenseInput.cs [hunk 2]
```diff
@@ -301,6 +314,28 @@ private void MultiMonthSave()
...
+                int weeksInRange = this.CalcWeeksInRange();
+                DateTime dtCurrentSaveDate = this.dtpStartDate.Value.Date;
+
+                //Loops for the amount of weeks in range
+                for (int weekIndex = 0; weekIndex < weeksInRange; weekIndex++)
+                {
+                    SaveNewExpense(dtCurrentSaveDate);
+
+                    //Set the date for the next expense
+                    dtCurrentSaveDate = dtCurrentSaveDate.AddMonths(1);
```
Comment (M-Zuber) [inline UI/RecurringExpenseInput.cs]: This seems like it should be AddDays(7)

File: UI/RecurringIncomeInput.cs [hunk 3]
```diff
@@ -246,6 +251,37 @@ private int CalcDaysInRange()
         /// Calculates the months in the range from the start date to the end date
         /// </summary>
         /// <returns>The number of months in the range</returns>
+        private int CalcWeeksInRange()
```
Comment (M-Zuber) [inline UI/RecurringIncomeInput.cs]: Same thing as for the expense form

File: UI/RecurringIncomeInput.cs [hunk 4]
```diff
@@ -246,6 +251,37 @@ private int CalcDaysInRange()
...
+                int weeksInRange = this.CalcWeeksInRange();
+                DateTime dtCurrentSaveDate = this.dtpStartDate.Value.Date;
+
+                //Loops for the amount of weeks in range
+                for (int weekIndex = 0; weekIndex < weeksInRange; weekIndex++)
+                {
+                    CreateNewIncome(dtCurrentSaveDate);
+
+                    //Set the date for the next expense
+                    dtCurrentSaveDate = dtCurrentSaveDate.AddMonths(1);
```
Comment (M-Zuber) [inline UI/RecurringIncomeInput.cs]: Same as expense form

**Comment by M-Zuber** [comment]
Thank you!
Overall it looks very nice, just some small details that need to be fixed.
I do still need to run some double checks before I merge

**Comment by kandoeniko** [comment]
Thanks for your feedback, I will correct the errors. It was my first
experience ever with github, I am still learning and I hope I didn't ruined
your project.

Have a nice day,
Eniko

2015-05-14 9:06 GMT+03:00 Mordechai Zuber <notifications@github.com>:

**Comment by M-Zuber** [comment]
@kandoeniko when you get a chance to fix the issues with this `PR`, please redo it onto the `Development` branch.
Thank you again, looking forwards to getting this code in.

**Comment by M-Zuber** [comment]
@kandoeniko any progress with this?

**Pull request closed without merging**